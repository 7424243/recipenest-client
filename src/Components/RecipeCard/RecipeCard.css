/* Classes */

.recipe-card {
    box-shadow: 0 4px 8px 0;
    transition: 0.3s;
    border-radius: 5px;
    border: 1px black solid;
    margin: 20px;
    padding: 2px;
    background-color: white;
    width: 175px;
    height: 275px;
    font-family: 'Baskervville', serif;
    font-size: medium;
}

.recipe-card-img {
    width: 100%;
    height: 175px;
    background-color: lightgray;
}

.recipe-card-name {
    padding-left: 3%;
    padding-right: 3%;
}
