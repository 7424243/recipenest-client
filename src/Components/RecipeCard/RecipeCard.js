import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import RecipenestContext from '../../RecipenestContext'
import PropTypes from 'prop-types'
import './RecipeCard.css'

class RecipeCard extends Component {

    //allow access to context
    static contextType = RecipenestContext

    render() {
        const {id, name, img_url} = this.props
        return (
            <Link to={`/recipe/${id}`}>
                <div className='recipe-card'>
                    <img className='recipe-card-img' src={img_url} alt='from recipe website' />
                    <h4 className='recipe-card-name'>{name}</h4>
                </div>
            </Link>
        )
    }
}

export default RecipeCard

RecipeCard.propTypes = {
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    img_url: PropTypes.string.isRequired,
}