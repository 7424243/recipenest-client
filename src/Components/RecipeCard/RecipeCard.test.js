import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import RecipeCard from './RecipeCard';

describe('RecipeCard Component', () => {

  //smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><RecipeCard/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<BrowserRouter><RecipeCard/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

})
