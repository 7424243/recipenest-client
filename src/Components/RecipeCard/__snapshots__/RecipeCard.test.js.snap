// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard Component renders the UI as expected 1`] = `
<a
  href="/recipe/1"
  onClick={[Function]}
>
  <div
    className="recipe-card"
  >
    <img
      alt="word cloud that describes the recipe"
      className="recipe-card-img"
      src="url"
    />
    <h4
      className="recipe-card-name"
    >
      name
    </h4>
  </div>
</a>
`;
